esp_flasher/gui.py

```
@@ -3,19 +3,416 @@
+        self.progress.setVisible(True)
+        improv = self._improv
+        threading.Thread(target=lambda: self._scan_bg(improv), daemon=True).start()
+
+    def _scan_bg(self, improv):
+        try:
+            networks = improv.request_wifi_networks()
+            # Sort by RSSI descending
+            networks.sort(key=lambda n: n[1], reverse=True)
+            # Thread-safe: emit signal to update UI on main thread
+            self._scan_finished.emit(networks)
+        except Exception as e:
```

Ruff (0.15.9)

[warning] Do not catch blind exception: `Exception`

(BLE001)